packages/core-ethereum/crates/core-ethereum-db/src/db.rs

```diff
@@ -518,15 +490,28 @@ impl<T: AsyncKVStorage<Key = Box<[u8]>, Value = Box<[u8]>>> HoprCoreEthereumDbAc
         let new_redeemed_balance = balance.add(&acked_ticket.ticket.amount);
         ops.put(key, new_redeemed_balance);
 
-        let key = utils_db::db::Key::new_with_prefix(counterparty, PENDING_TICKETS_COUNT)?;
-        let pending_balance = self
-            .db
-            .get_or_none::<Balance>(key.clone())
-            .await?
-            .unwrap_or(Balance::zero(BalanceType::HOPR));
-
-        let new_pending_balance = pending_balance.sub(&acked_ticket.ticket.amount);
-        ops.put(key, new_pending_balance);
+        if let Some(counterparty) = self.get_channel(&acked_ticket.ticket.channel_id).await?.map(|c| {
+            if c.source == self.me {
+                c.destination
+            } else {
+                c.source
+            }
+        }) {
+            let key = utils_db::db::Key::new_with_prefix(&counterparty, PENDING_TICKETS_COUNT)?;
+            let pending_balance = self
+                .db
+                .get_or_none::<Balance>(key.clone())
+                .await?
+                .unwrap_or(Balance::zero(BalanceType::HOPR));
+
+            let new_pending_balance = pending_balance.sub(&acked_ticket.ticket.amount);
+            ops.put(key, new_pending_balance);
+        } else {
+            error!(
+                "could not update redeemed tickets count: unable to find channel with id {}",
+                acked_ticket.ticket.channel_id
+            )
+        }
```

Comment:
It'd be beneficial, if we could wrap the entire set of operations in a "transaction", too many gets with possibility of anything heppening in-between.

Reply:
From the top level, the write lock must be already acquired to call this fn. From the bottom level, sure, but have we got a notion of transaction with the current DB abstraction?